### ISSUE TITLE
Package versioning and optional tf

### ISSUE THREAD

**Issue by simonrnss** [AI-SDC/AI-SDC#139]
AISDC requirements has very specific (and potentially problematic) package versions. This looks to cause problems on Jim's laptop (and probably elsewhere). It also takes a long time to install.

This is a placeholder for now.

## Jim's mac install issue

@simonrnss can test on a macbook pro
@rpreen can test on a macos runner

## Removing tensorflow

@rpreen to look into providing option to install AISDC without tf support as per openai gym

**Comment by rpreen** [comment]
I can't reproduce the macOS install problem.

CI runner using Python 3.10.11 works: https://github.com/AI-SDC/AI-SDC/actions/runs/4930955186/jobs/8812495760

**Comment by simonrnss** [comment]
Same here -- install was fine on my macbook running py 3.10

**Comment by rpreen** [comment]
The safemodel package (within aisdc) depends on tensorflow so if it's made optional then you would be able to import and run the attacks package fine, but you can't import and use any safemodel classifiers (e.g., `from aisdc.safemodel.classifiers import SafeRandomForestClassifier` will access the [`__init__.py`](https://github.com/AI-SDC/AI-SDC/blob/development/aisdc/safemodel/classifiers/__init__.py), which imports all the safemodel classifiers, including keras. The reason for structuring the safemodel classifiers in a package with that init file is to make the imports nice and clean for the end user.

So basically that means we could make safemodels an optional part of the aisdc package in order to make tensorflow optional, but is this worth doing?

**Comment by rpreen** [comment]
Possibly one option is to further sub-package the safe models. So instead of putting them all in `aisdc.safemodel.classifiers` we could have `aisdc.safemodel.sklearn.classifiers` and `aisdc.safemodel.tf.classifiers` and break apart the sklearn and tensorflow safe models.  Of course this means the researchers then importing the respective classifier will have to differentiate between them on the basis of sklearn and tf.

**Comment by simonrnss** [comment]
My input, fwiw, is that this would probably become a time-drain that we don't really have capacity for at the moment. The time to install tf is a pain, but not as big a pain as it looks like removing it might be!
